system/ee/legacy/database/DB_active_rec.php

```diff
@@ -302,15 +303,21 @@ public function join($table, $cond, $type = '')
         $this->_track_aliases($table);
 
         // Strip apart the condition and protect the identifiers
-        if (preg_match('/([\w\.]+)([\W\s]+)(.+)/', $cond, $match)) {
+        if (preg_match('/([\w\.]+)([\W\s]+)(.+)/', $cond, $match) && !$alias) {
             $match[1] = $this->_protect_identifiers($match[1]);
             $match[3] = $this->_protect_identifiers($match[3]);
 
             $cond = $match[1] . $match[2] . $match[3];
         }
 
+        // If a join alias specified, extract it now
+        $join_alias = '';
+        if ($alias) {
+            $join_alias = ' '.$alias.' ';
+        }
```

Comment:
Can you add spaces around `.` to make the code PSR-12? Also, since `$alias` is never a boolean, it is probably more foolproof to check it with `is_empty` (thinking of future PHP versions here)